{
    "n_steps": 22,
    "cum_reward": 0,
    "cum_raw_reward": 0,
    "err_msg": "Exception uncaught by agent or environment in task workarena.servicenow.order-apple-mac-book-pro15.\nError:\nPage.evaluate: Execution context was destroyed, most likely because of a navigation",
    "stack_trace": "Traceback (most recent call last):\n  File \"/Users/patrick/Dropbox/PARA/20_code/2024-12-09_WorkBench/BrowserGym_n/.venv/lib/python3.12/site-packages/browsergym/experiments/loop.py\", line 276, in run\n    step_info.from_step(env, action, obs_preprocessor=agent.obs_preprocessor)\n  File \"/Users/patrick/Dropbox/PARA/20_code/2024-12-09_WorkBench/BrowserGym_n/.venv/lib/python3.12/site-packages/browsergym/experiments/loop.py\", line 411, in from_step\n    self.obs, self.reward, self.terminated, self.truncated, env_info = env.step(action)\n                                                                       ^^^^^^^^^^^^^^^^\n  File \"/Users/patrick/Dropbox/PARA/20_code/2024-12-09_WorkBench/BrowserGym_n/.venv/lib/python3.12/site-packages/gymnasium/wrappers/common.py\", line 125, in step\n    observation, reward, terminated, truncated, info = self.env.step(action)\n                                                       ^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/patrick/Dropbox/PARA/20_code/2024-12-09_WorkBench/BrowserGym_n/.venv/lib/python3.12/site-packages/gymnasium/wrappers/common.py\", line 393, in step\n    return super().step(action)\n           ^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/patrick/Dropbox/PARA/20_code/2024-12-09_WorkBench/BrowserGym_n/.venv/lib/python3.12/site-packages/gymnasium/core.py\", line 322, in step\n    return self.env.step(action)\n           ^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/patrick/Dropbox/PARA/20_code/2024-12-09_WorkBench/BrowserGym_n/.venv/lib/python3.12/site-packages/browsergym/core/env.py\", line 433, in step\n    reward, done, user_message, task_info = self._task_validate()\n                                            ^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/patrick/Dropbox/PARA/20_code/2024-12-09_WorkBench/BrowserGym_n/.venv/lib/python3.12/site-packages/browsergym/core/env.py\", line 458, in _task_validate\n    reward, done, user_message, info = self.task.validate(self.page, self.chat.messages)\n                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/patrick/Dropbox/PARA/20_code/2024-12-09_WorkBench/BrowserGym_n/.venv/lib/python3.12/site-packages/browsergym/workarena/tasks/service_catalog.py\", line 475, in validate\n    right_url = check_url_suffix_match(page, expected_url=self.final_url, task=self)\n                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/patrick/Dropbox/PARA/20_code/2024-12-09_WorkBench/BrowserGym_n/.venv/lib/python3.12/site-packages/browsergym/workarena/tasks/utils/utils.py\", line 14, in check_url_suffix_match\n    page_url = page.evaluate(\"window.location.href\")\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/patrick/Dropbox/PARA/20_code/2024-12-09_WorkBench/BrowserGym_n/.venv/lib/python3.12/site-packages/playwright/sync_api/_generated.py\", line 8355, in evaluate\n    self._sync(\n  File \"/Users/patrick/Dropbox/PARA/20_code/2024-12-09_WorkBench/BrowserGym_n/.venv/lib/python3.12/site-packages/playwright/_impl/_sync_base.py\", line 115, in _sync\n    return task.result()\n           ^^^^^^^^^^^^^\n  File \"/Users/patrick/Dropbox/PARA/20_code/2024-12-09_WorkBench/BrowserGym_n/.venv/lib/python3.12/site-packages/playwright/_impl/_page.py\", line 440, in evaluate\n    return await self._main_frame.evaluate(expression, arg)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/patrick/Dropbox/PARA/20_code/2024-12-09_WorkBench/BrowserGym_n/.venv/lib/python3.12/site-packages/playwright/_impl/_frame.py\", line 278, in evaluate\n    await self._channel.send(\n  File \"/Users/patrick/Dropbox/PARA/20_code/2024-12-09_WorkBench/BrowserGym_n/.venv/lib/python3.12/site-packages/playwright/_impl/_connection.py\", line 59, in send\n    return await self._connection.wrap_api_call(\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/patrick/Dropbox/PARA/20_code/2024-12-09_WorkBench/BrowserGym_n/.venv/lib/python3.12/site-packages/playwright/_impl/_connection.py\", line 514, in wrap_api_call\n    raise rewrite_error(error, f\"{parsed_st['apiName']}: {error}\") from None\nplaywright._impl._errors.Error: Page.evaluate: Execution context was destroyed, most likely because of a navigation\n",
    "stats.cum_steps": 23,
    "stats.cum_n_token_last_action": 3141,
    "stats.max_n_token_last_action": 188,
    "stats.cum_n_token_last_action_error": 154,
    "stats.max_n_token_last_action_error": 154,
    "stats.cum_n_token_axtree_txt": 778126,
    "stats.max_n_token_axtree_txt": 86675,
    "stats.cum_n_token_pruned_html": 2829492,
    "stats.max_n_token_pruned_html": 287570,
    "stats.cum_step_elapsed": 170.15504217147827,
    "stats.max_step_elapsed": 23.117121934890747,
    "stats.cum_agent_elapsed": 106.32049369812012,
    "stats.max_agent_elapsed": 14.293729782104492,
    "terminated": null,
    "truncated": null
}